apm-sniffer/apm-agent-core/src/main/java/org/skywalking/apm/agent/core/boot/DefaultNamedThreadFactory.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2017, OpenSkywalking Organization All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Project repository: https://github.com/OpenSkywalking/skywalking
+ */
+package org.skywalking.apm.agent.core.boot;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * @author zhangkewei
+ */
+public class DefaultNamedThreadFactory implements ThreadFactory {
+    private static final AtomicInteger BOOT_SERVICE_SEQ = new AtomicInteger(0);
+    private final AtomicInteger threadSeq = new AtomicInteger(0);
+    private final String namePrefix;
+    private final ThreadGroup group;
+    public DefaultNamedThreadFactory(String name) {
+        namePrefix = "SkywalkingAgent-" + BOOT_SERVICE_SEQ.incrementAndGet() + "-" + name + "-";
+        SecurityManager s = System.getSecurityManager();
+        group = (s != null) ? s.getThreadGroup() : Thread.currentThread().getThreadGroup();
+    }
+    @Override
+    public Thread newThread(Runnable r) {
+        Thread t = new Thread(group, r,
+                namePrefix + threadSeq.getAndIncrement(),
+                0);
+        if (t.isDaemon())
+            t.setDaemon(false);
```

Comment:
Setting `Thread#isDaemon` to **false**, may cause JVM can't exit, if agent's thread is still running. IMHO, you should set to **true** as default. BTW, **false** is the default value of any thread.